add quarantine_size_mb (#3264)
ROCm/ROCm: docs/conceptual/using-gpu-sanitizer.md

There is an environment variable, `ASAN_OPTIONS`, that can be used to adjust the runtime behavior of the ASan runtime itself. There are more than a hundred "flags" that can be adjusted (see an old list at [flags](https://github.com/google/sanitizers/wiki/AddressSanitizerFlags)) but the default settings are correct and should be used in most cases. It must be noted that these options only affect the host ASan runtime. The device runtime only currently supports the default settings for the few relevant options.

> There are three `ASAN_OPTION` flags of particular note.

Author: There are three `ASAN_OPTION` flags of note.

There is an environment variable, `ASAN_OPTIONS`, that can be used to adjust the runtime behavior of the ASan runtime itself. There are more than a hundred "flags" that can be adjusted (see an old list at [flags](https://github.com/google/sanitizers/wiki/AddressSanitizerFlags)) but the default settings are correct and should be used in most cases. It must be noted that these options only affect the host ASan runtime. The device runtime only currently supports the default settings for the few relevant options.

> There are three `ASAN_OPTION` flags of particular note.

Reviewer: ```suggestion
There are three `ASAN_OPTION` flags of note.
```

> This option directs the ASan runtime to enable the [Leak Sanitizer](https://clang.llvm.org/docs/LeakSanitizer.html) (LSan). Unfortunately, for heterogeneous applications, this default will result in significant output from the leak sanitizer when the application exits due to allocations made by the language runtime which are not considered to be leaks. This output can be avoided by adding `detect_leaks=0` to the `ASAN_OPTIONS`, or alternatively by producing an LSan suppression file (syntax described [here](https://github.com/google/sanitizers/wiki/AddressSanitizerLeakSanitizer)) and activating it with environment variable `LSAN_OPTIONS=suppressions=/path/to/suppression/file`. When using a suppression file, a suppression report is printed by default. The suppression report can be disabled by using the `LSAN_OPTIONS` flag `print_suppressions=0`.

Reviewer: ```suggestion
This option directs the ASan runtime to enable the [Leak Sanitizer](https://clang.llvm.org/docs/LeakSanitizer.html) (LSan). For heterogeneous applications, this default results in significant output from the leak sanitizer when the application exits due to allocations made by the language runtime which are not considered to be leaks. This output can be avoided by adding `detect_leaks=0` to the `ASAN_OPTIONS`, or alternatively by producing an LSan suppression file (syntax described [here](https://github.com/google/sanitizers/wiki/AddressSanitizerLeakSanitizer)) and activating it with environment variable `LSAN_OPTIONS=suppressions=/path/to/suppression/file`. When using a suppression file, a suppression report is printed by default. The suppression report can be disabled by using the `LSAN_OPTIONS` flag `print_suppressions=0`.
```

* `quarantine_size_mb=N default 256`

> This option defines the number of megabytes (MB) `N` of memory that the ASAN runtime will hold after it is `free`d to detect use-after-free situations.

Reviewer: The ASAN occurrence here is all uppercase and not ASan

Author: I made them consistent with ASan

> This memory is unavailable for other purposes. The default of 256 MB may be too small to detect some use-after-free situations, especially given that the large size of many GPU memory allocations may push `free`d allocations out of quarantine before the attempted use.

Reviewer: ```suggestion
This memory is unavailable for other purposes. The default of 256 MB might be too small to detect some use-after-free situations, especially given that the large size of many GPU memory allocations might push `free`d allocations out of quarantine before the attempted use.
```

Reviewer: i don't understand free'd

Does this make sense?
...many GPU memory allocations might push allocations that are now free out of quarantine before the attempted use.

Author: I sent you a message regarding freed.
